DhsTranslator/src/DhsService.h

```diff
@@ -0,0 +1,333 @@
+                                axiom_node_t *itemValNode =
+                                        axiom_node_get_first_child(itemNode,
+                                                env);
+                                boost::optional<T> val = parseVal<T>(env,
+                                        itemValNode);
+                                if (val) {
+                                    array.push_back(val.get());
+                                } else {
+                                    ret = false;
+                                }
+                            } else {
+                                ret = false;
+                            }
+                        } else {
+                            ret = false;
+                        }
+                    }
+                } else {
+                    ret = false;
+                }
+            } else {
+                ret = false;
+            }
+            axutil_qname_free(typeAttrName, env);
+        } else {
+            ret = false;
+        }
+        return ret;
+    }
+
+    bool parseArrayNode(const axutil_env_t *env, axiom_node_t *node,
+            std::vector<int> &array) {
+        return parseArrayNode(env, node, array, "int");
+    }
+    bool parseArrayNode(const axutil_env_t *env, axiom_node_t *node,
+            std::vector<std::string> &array) {
+        return parseArrayNode(env, node, array, "string");
+    }
+    bool parseArrayNode(const axutil_env_t *env, axiom_node_t *node,
+            std::vector<axiom_node_t *> &array) {
+        return parseArrayNode(env, node, array, "object");
+    }
+    bool parseArrayNode(const axutil_env_t *env, axiom_node_t *node,
+            std::vector<IDhsAdapter::Keyword> &array) {
+        return parseArrayNode(env, node, array, "object");
+    }
+
+public:
+    virtual ~DhsService();
+    static axis2_svc_skeleton_t* create(const axutil_env_t *env);
+    static axis2_status_t remove(axis2_svc_skeleton_t *svc_skeleton,
+            const axutil_env_t *env);
+
+    static int stubInit(axis2_svc_skeleton_t *svc_skeleton,
+            const axutil_env_t *env);
+    static int stubInitWithConfig(axis2_svc_skeleton_t *svc_skeleton,
+            const axutil_env_t *env, struct axis2_conf * conf);
+    static axiom_node_t* stubInvoke(axis2_svc_skeleton_t *svc_skeli,
+            const axutil_env_t *env, axiom_node_t *node,
+            axis2_msg_ctx_t *msg_ctx);
+    static axiom_node_t* stubOnFault(axis2_svc_skeleton_t *svc_skeli,
+            const axutil_env_t *env, axiom_node_t *node);
+    static axis2_status_t stubFree(axis2_svc_skeleton_t *svc_skeli,
+            const axutil_env_t *env);
+
+    const IDhsAdapter* getDhsAdapter() const {
+        return dhsAdapter;
+    }
+
+    void setDhsAdapter(IDhsAdapter* dhsAdapter) {
+        this->dhsAdapter = dhsAdapter;
+    }
+};
+
+//Type bool needs an specialized version because it needs to enable alpha values in the conversion.
```

Comment:
In this context, the string representation of the boolean values `true` and `false`. By default a stream only converts "0" and "1" to boolean values.